File: typescript/examples/get-player-stats.ts
import { MythicMC, NotFoundError } from '../src/index.ts'

const api = new MythicMC({ apiKey: process.env.MYTHICMC_API_KEY!, baseUrl: process.env.MYTHICMC_API_URL })
const player = process.argv[2] ?? 'Vicente_1313'

try {
  const { survival } = await api.getPlayerStats(player)

  const { kills, deaths } = survival.combat
  console.log(`kills ${kills}, deaths ${deaths}, K/D ${(kills / Math.max(deaths, 1)).toFixed(2)}`)
  console.log('blocks mined:', survival.world.blocksMined)

  // Missing groups are unknown, not zero.
  console.log('net worth:', survival.netWorth ? `$${survival.netWorth.total} (#${survival.netWorth.rank})` : 'unknown')
  console.log('balance:', survival.money ?? 'unknown')
  console.log('events won:', survival.events ? survival.events.wins.bingo + survival.events.wins.raffle : 'unknown')
} catch (error) {
  if (!(error instanceof NotFoundError)) throw error
  console.log(`no Survival statistics for ${player}: ${error.message}`)
}

File: python/examples/get_player_stats.py
import os
import sys

from mythicmc import DEFAULT_BASE_URL, MythicMC, NotFoundError

key = os.environ["MYTHICMC_API_KEY"]
base_url = os.environ.get("MYTHICMC_API_URL", DEFAULT_BASE_URL)
player = sys.argv[1] if len(sys.argv) > 1 else "Vicente_1313"

try:
    with MythicMC(key, base_url=base_url) as api:
        survival = api.get_player_stats(player).survival
except NotFoundError as error:
    sys.exit(f"no Survival statistics for {player}: {error.message}")

combat = survival.combat
print(f"kills {combat.kills}, deaths {combat.deaths}, K/D {combat.kills / max(combat.deaths, 1):.2f}")

# Missing groups are unknown, not zero.
net_worth = survival.net_worth
print("net worth:", f"${net_worth.total} (#{net_worth.rank})" if net_worth else "unknown")
print("balance:", survival.money if survival.money is not None else "unknown")
events = survival.events
print("events won:", events.wins.bingo + events.wins.raffle if events else "unknown")
